arroyo/processing/strategies/transform.py

```diff
@@ -65,14 +65,18 @@ def poll(self) -> None:
     def submit(self, message: Message[TPayload]) -> None:
         assert not self.__closed
 
-        self.__next_step.submit(
-            Message(
-                message.partition,
-                message.offset,
-                self.__transform_function(message),
-                message.timestamp,
+        result = self.__transform_function(message)
+        if isinstance(message.data, BrokerPayload):
+            broker_payload = BrokerPayload(
+                result,
+                message.data.partition,
+                message.data.offset,
+                message.data.timestamp,
             )
-        )
+            self.__next_step.submit(Message(broker_payload))
+        else:
+            payload = Payload(result, message.committable)
+            self.__next_step.submit(Message(payload))
```

Comment:
Same issue I had above, this looks the same pattern. Transform the payload of `Payload` or `BrokerPayload` and package it into a message. We can put it inside the payload class so we do not replciate this pattern everywhere.

Reply:
Updated